src/text.ts

```diff
@@ -45,7 +41,8 @@ export function setTextContent(node: Node, text: string): void {
   } else if (isTextNode(node)) {
     node.value = text;
   } else if (isParentNode(node)) {
-    const textNode = createTextNode(text);
-    appendChild(node, textNode);
+    var parent: ParentNode = node;
```

Comment:
this should be a `const`, but actually i think you should be able to lose this variable anyway.

the `isParentNode` above already narrows the type, meaning `node.childNodes` should exist already (i.e. you can do `node.childNodes = [...]`)

Reply:
I thought I was getting a type error before so I did that, I must have been referencing some other variable.

I noticed that some of the tests I wrote were failing because the `parentNode` value was `null`. I think this comes from no longer using the `appendChild` method from the tree adapter. Should I make the `createTextNode` method take in a parameter for the parent node or should I go back to using the append child method by doing something like this?
```ts
node.childNodes = [];
appendChild(node, createTextNode(text));
```

Reply:
i would do what you said:

```ts
node.childNodes = [];
appendChild(node, createTextNode(text));
```

at least means we only have one place that appends nodes, too.